File: maxAreaOfWater/maxarea.ts
function maxArea(height: number[]): number {
    let left = 0
    let right = height.length - 1
    let maxArea = 0;

    while(left < right) {
        let minHeight = Math.min(height[left],height[right])
        let width = right - left

        let currentArea = width * minHeight

        if (currentArea > maxArea) {
            maxArea = currentArea
        }

        if(height[left] < height[right]) {
            left++
        }
        else {
            right--
        }
    }
    return maxArea
};

File: maxAreaOfWater/maxarea.go
package maxareaofwater

func maxArea(height []int) int {
	left, right, maxArea := 0, len(height)-1, 0

	for left < right {

		minHeight := height[left]

		if height[left] > height[right] {
			minHeight = height[right]
		}
		width := right - left
		currentArea := width * minHeight

		if currentArea > maxArea {
			maxArea = currentArea
		}
		if height[left] < height[right] {
			left++
		} else {
			right++
		}
	}
	return maxArea

}
